packages/devtools_server/lib/src/handlers.dart

```diff
@@ -103,3 +177,182 @@ class ServerApi {
   FutureOr<shelf.Response> notImplemented(shelf.Request request) =>
       shelf.Response(204);
 }
+
+// Access the ~/.flutter file.
+class Usage {
+  Analytics _analytics;
+
+  /// Create a new Usage instance; [versionOverride] and [configDirOverride] are
+  /// used for testing.
+  Usage(
+      {String settingsName = 'flutter',
+      String versionOverride,
+      String configDirOverride}) {
+    // final FlutterVersion flutterVersion = FlutterVersion.instance;
+    // final String version = versionOverride ?? flutterVersion.getVersionString(redactUnknownBranches: true);
+    // TODO(terry): UA, first parameter, is '' could be DevTools UA
+    // TODO(terry): version, second parameter, is '' could be real Flutter version #.
+    // TODO(terry): documentDirectory, third parameter, is null could be :
+    //    documentDirectory: configDirOverride != null ? fs.directory(configDirOverride) : null
+    _analytics = AnalyticsIO('', settingsName, '', documentDirectory: null);
+  }
+
+  /// Does the .flutter store exist?
+  static bool get doesStoreExits {
+    final flutterStore = File('${DevToolsUsage.userHomeDir()}/.flutter');
+    return flutterStore.existsSync();
+  }
+
+  bool get isFirstRun => _analytics.firstRun;
+
+  bool get enabled => _analytics.enabled;
+
+  set enabled(bool value) => _analytics.enabled = value;
+
+  String get clientId => _analytics.clientId;
+}
+
+// Access the DevTools on disk store (~/.devtools).
+class DevToolsUsage {
```

Comment:
Add server side tests of the APIs to modify DevToolsUsage. This code should be easy to test and it would be nice to verify that we don't have a bug causing us to send surveys to users multiple times or crash running the code on windows.

Reply:
TBD